File: python/gestorAplicacion/gestorVuelos/aeropuerto.py
class Aeropuerto:
    def __init__(self,nombre=None):
        self._nombre=nombre
        self._vuelos=[]
        self._pasajeros=[]
        self._empleados=[]
        self._aviones=[]
        self._transaccionesKeys=[]
        self._transaccionesValues=[]
        self._dinero=0

    def getNombre(self):
        return self._nombre

    def setNombre(self,nombre):
        self._nombre=nombre

    def getVuelos(self):
        return self._vuelos

    def setVuelos(self,vuelos):
        self._vuelos=vuelos

    def addVuelo(self,vuelo):
        self._vuelos.append(vuelo)

    def getPasajeros(self):
        return self._pasajeros

    def setPasajeros(self,pasajeros):
        self._pasajeros=pasajeros

    def addPasajero(self,pasajeros):
        self._pasajeros.append(pasajeros)

    def getEmpleados(self):
        return self._empleados

    def setEmpleados(self,empleados):
        self._empleados=empleados

    def addEmpleado(self,empleados):
        self._empleados.append(empleados)

    def getAviones(self):
        return self._aviones

    def setAviones(self,aviones):
        self._aviones=aviones

    def addAvion(self,aviones):
        self._aviones.append(aviones)

    def getTransaccionesKeys(self):
        return self._transaccionesKeys

    def setTransaccionesKeys(self,transaccionesKeys):
        self._transaccionesKeys=transaccionesKeys

    def addTransaccionesKeys(self,transaccionesKeys):
        self._transaccionesKeys.append(transaccionesKeys)

    def getTransaccionesValues(self):
        return self._transaccionesValues

    def setTransaccionesValues(self,transaccionesValues):
        self._transaccionesValues=transaccionesValues

    def addTransaccionesValues(self,transaccionesValues):
        self._transaccionesValues.append(transaccionesValues)

    def getDinero(self):
        return self._dinero

    def setDinero(self,dinero):
        self._dinero=dinero

    def buscarEmpleado(self,id):
        for i in self._empleados:
            if i.getCedula()==id:
                return i
    def buscarPasajero(self,id):
        for i in self._pasajeros:
            if i.getCedula()==id:
                return i
    def ingresarDinero(self, pago):
        self._dinero += pago

    def transaccion(self, concepto, valor):
        self._transaccionesKeys.append(concepto)
        self._transaccionesValues.append(valor)
        self.ingresarDinero(valor)

    def buscarVuelo(self,sala,modelo,destino):
        for i in self._vuelos:
            if i.getDestino()==destino and i.getAvion().getModelo()==modelo and i.getSalaEmbarque()==sala:
                return i

    def buscarAvion(self,modelo):
        for i in self._aviones:
            if i.getModelo()==modelo:
                return i

    def despedirEmpleado(self,empleado):
        for i in self._empleados:
            if i==empleado:
                try:
                    empleado.getVuelo().remove(empleado)
                except:
                    pass
                self._empleados.remove(empleado)

    def eliminarPasajero(self,pasajero):
        for i in self._pasajeros:
            if i==pasajero:
                try:
                    pasajero.getVuelo().remove(pasajero)
                except:
                    pass
                self._pasajeros.remove(pasajero)

    def cancelarVuelo(self,vuelo):
        for i in self._vuelos:
            if i==vuelo:
                try:
                    vuelo.getAvion().remove(vuelo)
                except:
                    pass
                self._vuelos.remove(vuelo)

    def cancelarAvion(self,avion):
        for i in self._aviones:
            if i==avion:
                try:
                    avion.getvuelo().remove(avion)
                except:
                    pass
                self._aviones.remove(avion)

File: src/gestorAplicacion/gestionVuelos/Aeropuerto.java
// Autores: Juan Jose Zapata Cadavid, Juan Camilo Molina Roncancio
/* Se encarga del manejo de los atributos del aeropuerto, como su nombre indica
 * es donde se van a almacenar todos los datos del sistema.
 */

package gestorAplicacion.gestionVuelos;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import basedatos.Deserializador;
import gestorAplicacion.gestionHumana.Empleado;
import gestorAplicacion.gestionHumana.Pasajero;

public class Aeropuerto implements Serializable, zonasEmbarque {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private String nombre;
	private List<Vuelo> vuelos = new ArrayList<>();
	private List<Empleado> empleados = new ArrayList<>();
	private List<Pasajero> pasajeros = new ArrayList<>();
	private List<Avion> aviones = new ArrayList<>();
	private List<String> transaccionesKeys = new ArrayList<>();
	private List<Integer> transaccionesValues = new ArrayList<>();
	private long dinero;

	public Aeropuerto() {
		Deserializador.deserializar(this);
	}

	public void ingresarDinero(int pago) {
		dinero += pago;
	}

	public long getDinero() {
		return dinero;
	}

	public void setDinero(long dinero) {
		this.dinero = dinero;
	}

	public Aeropuerto(String nombre) {
		this.nombre = nombre;
	}

	public void agregarVuelo(Vuelo vuelo) {
		this.vuelos.add(vuelo);
	}

	public void agregarEmpleado(Empleado empleado) {
		this.empleados.add(empleado);
	}

	public void agregarAvion(Avion avion) {
		this.aviones.add(avion);
	}

	public List<Vuelo> getVuelos() {
		return vuelos;
	}

	public void setVuelos(List<Vuelo> vuelos) {
		this.vuelos = vuelos;
	}

	public List<Empleado> getEmpleados() {
		return empleados;
	}

	public void setEmpleados(List<Empleado> empleados) {
		this.empleados = empleados;
	}

	public List<Avion> getAviones() {
		return aviones;
	}

	public void setAviones(List<Avion> aviones) {
		this.aviones = aviones;
	}

	public List<Pasajero> getPasajeros() {
		return pasajeros;
	}

	public void setPasajeros(List<Pasajero> pasajeros) {
		this.pasajeros = pasajeros;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public void despedirEmpleado(Empleado empleado) {
		if (empleado.getVuelo() != null) {
			empleado.getVuelo().getEmpleados().remove(empleado);
		}

		empleados.remove(empleados.indexOf(empleado));
	}

	/*
	 * metodo para realizar una transaccion primer argumento es un String con el
	 * concepto (breve descripcion) de la transaccion segundo argumento es el valor
	 * de la transaccion (si es retiro de dinero debe ser negativo) el metodo retira
	 * el dinero correspondiente del aeropuerto y añade la transaccion a la lista
	 */
	public void transaccion(String concepto, int valor) {
		transaccionesKeys.add(concepto);
		transaccionesValues.add(valor);

		ingresarDinero(valor);
	}

	// Metodo para desplegar el historial de transacciones al usuario
	public String transacciones() {
		int acumulador = 0;
		StringBuilder historial = new StringBuilder();

		for (int i = 0; i < Math.min(transaccionesKeys.size(), transaccionesValues.size()); i++) {
			historial.append(i + 1).append(":  ").append(transaccionesKeys.get(i)).append(" --- ")
					.append(transaccionesValues.get(i)).append("\n");
			acumulador += transaccionesValues.get(i);
		}
		historial.append("rendimiento total : --- ").append(acumulador);
		return historial.toString();
	}

	public List<String> getTransaccionesKeys() {
		return transaccionesKeys;
	}

	public void setTransaccionesKeys(List<String> transaccionesKeys) {
		this.transaccionesKeys = transaccionesKeys;
	}

	public List<Integer> getTransaccionesValues() {
		return transaccionesValues;
	}

	public void setTransaccionesValues(List<Integer> transaccionesValues) {
		this.transaccionesValues = transaccionesValues;
	}
}
